src/http/ut-http-message-decoder.c

```diff
@@ -122,6 +122,11 @@ static char *get_string(UtObject *data, size_t start, size_t end) {
   return ut_string_take_text(string);
 }
 
+static bool valid_http_version(ut_cstring_ref protocol_version) {
```

Comment:
This should be 'const char*' - I'm surprised doing it the way you had even compiles or doesn't do a double free!?